cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -38,40 +40,34 @@ pub struct Registry {
     _json: Value,
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
-pub struct LockedManifest(Value);
-
-impl LockedManifest {
-    /// Use pkgdb to lock a manifest
-    ///
-    /// `existing_lockfile_path` can be either the global lock or an environment's
-    /// lockfile
-    pub fn lock_manifest(
-        pkgdb: &Path,
-        manifest_path: &Path,
-        existing_lockfile_path: &CanonicalPath,
-        global_manifest_path: &Path,
-    ) -> Result<Self, LockedManifestError> {
-        let canonical_manifest_path =
-            CanonicalPath::new(manifest_path).map_err(LockedManifestError::BadManifestPath)?;
-
-        let mut pkgdb_cmd = Command::new(pkgdb);
-        pkgdb_cmd
-            .args(["manifest", "lock"])
-            .arg("--ga-registry")
-            .arg("--global-manifest")
-            .arg(global_manifest_path)
-            .arg("--manifest")
-            .arg(canonical_manifest_path)
-            .arg("--lockfile")
-            .arg(existing_lockfile_path);
+#[derive(Debug, Clone, Serialize)]
+#[serde(untagged)]
+pub enum LockedManifest {
+    Catalog(LockedManifestCatalog),
+    Pkgdb(LockedManifestPkgdb),
+}
 
-        debug!("locking manifest with command: {}", pkgdb_cmd.display());
-        call_pkgdb(pkgdb_cmd)
-            .map_err(LockedManifestError::LockManifest)
-            .map(Self)
+impl<'de> Deserialize<'de> for LockedManifest {
+    fn deserialize<D>(deserializer: D) -> Result<LockedManifest, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = Value::deserialize(deserializer)?;
+        let version = value.get("lockfile-version").and_then(Value::as_u64);
+
+        match version {
+            Some(0) => Ok(LockedManifest::Pkgdb(LockedManifestPkgdb(value))),
```

Comment:
suggestion: handle `None` the same as we handle it in `RawManifest::to_typed`. `None => Pkgdb, Some(1) => Catalog, Some(_) => Err`

Reply:
Ah I had missed that, looks good as is